src/middleware/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { AuthenticationError } from "apollo-server";
+import { logger } from "../logger";
+import * as dotenv from "dotenv";
+
+dotenv.config();
```

Comment:
I recommend moving this to the top of your entry point file where the server is initialized. Currently, all your services and middleware that rely on environment variables are dependent on the initialization of the authentication middleware.

Reply:
Well noted, I didn't think of that. I moved this to the top of app.ts